src/index.ts

```diff
@@ -77,18 +77,44 @@ export function identity<B extends AnyBrand>(underlying: BaseOf<B>): B {
 }
 
 /**
- * Produces a `Brander<B>`, given a brand type `B`. This simply returns
- * `identity` but relies on type inference to give the return type the correct
- * type.
+ * Asserts that a value is the type `Output`.
+ */
+export type AssertionFunction<Input, Output extends Input> = (
+  value: Input,
```

Comment:
It would have been nice to make this `: unknown`. But, then validator functions with typed params such `value: number` in the tests wouldn't be assignable. Unless I'm missing some clever technique I think the choices are:

* Have `value` always be `unknown` and only have one type parameter `T`/`Output` _(original hope)_
* Have `value`'s type by generic, so there are two type parameters _(this)_

Since the branders returned by `make()` take in a single argument that's the base type of the brand, I figured the latter was better for users.